src/v/cluster/drain_manager.cc

```diff
@@ -197,7 +198,10 @@ ss::future<> drain_manager::do_drain() {
         std::vector<ss::future<std::error_code>> transfers;
         transfers.reserve(selected.size());
         for (auto& p : selected) {
-            transfers.push_back(p->transfer_leadership(std::nullopt));
+            auto req = transfer_leadership_request{
+              .group = p->group(),
```

Comment:
This is what originally confused me: this assignment is not necessary, right? We could just as well leave it as 0?

Reply:
Yeah it definitely could. Since we do pass it though I figured it should just be as accurate as we can make it. In case someone tries reading it down the line.